### ISSUE TITLE
Type 'boolean' is not assignable to type 'Expression' in .aggregate()

### ISSUE THREAD

**Issue by DavideViolante** [Automattic/mongoose#11980]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.4.0 (happening also in 6.3.9)

### Node.js version

16.15.1

### MongoDB server version

idk

### Description

After upgrading from v6.3.8 to v6.4.0 I've the following error for TS. I couldnt find another similar issue, sorry if it exists already:
TS version v4.6.2
```
No overload matches this call.
  Overload 1 of 2, '(pipeline?: PipelineStage[], options?: AggregateOptions, callback?: Callback<any[]>): Aggregate<any[]>', gave the following error.
    Type 'boolean' is not assignable to type 'Expression'.
  Overload 2 of 2, '(pipeline: PipelineStage[], callback?: Callback<any[]>): Aggregate<any[]>', gave the following error.
    Type 'boolean' is not assignable to type 'Expression'.ts(2769)
```

### Steps to Reproduce

```ts
    const clients = await Client.aggregate([
      { $match: { active: true } }, // active is underlined for error
    ]);
```

In the schema active is type: Boolean.

**Comment by ilhanbozcan** [comment]
Basically, you can add a boolean type to expression. Check https://github.com/Automattic/mongoose/pull/11974

**Comment by ayoubyt** [comment]
I have a similar case, it's just now with "ObjectId" type, for example. code in this method gives a typing error
![image](https://user-images.githubusercontent.com/36169432/175394172-76855501-98e7-4dcc-9b78-b51e052be7c2.png)
```
Overload 1 of 2, '(pipeline?: PipelineStage[], options?: AggregateOptions, callback?: Callback<any[]>): Aggregate<any[]>', gave the following error.
    Type 'ObjectId' is not assignable to type 'Expression | Record<string, Expression>'.
      Type 'ObjectId' is not assignable to type 'Record<string, Expression>'.
        Index signature for type 'string' is missing in type 'ObjectId'.
  Overload 2 of 2, '(pipeline: PipelineStage[], callback?: Callback<any[]>): Aggregate<any[]>', gave the following error.
    Type 'ObjectId' is not assignable to type 'Expression | Record<string, Expression>'.

43         $match: new Types.ObjectId("62b4cd0ec71e68ed41fbc427")
```
highlighting  ```Type 'ObjectId' is not assignable to type 'Expression | Record<string, Expression>'.```

so is that okey and my code is wrong and can be changed ?, or it's an issue with the latest Mongoose releases ?

**Comment by andreialecu** [comment]
This should be fixed by #11969
